src/main/kotlin/net/emdal/tank/Graph.kt

```diff
@@ -41,24 +42,25 @@ data class Graph(val query: String = "", val aliases: List<Pair<String, Entity?>
   ): Graph {
     return this.copy(
       query = query + "-[$alias${block().body()}]->",
-      aliases = aliases + (alias to null)
+      aliases = aliases + (alias to emptyList())
     )
   }
 
   fun <T : Relationship> relationship(
     alias: String = "",
-    madeFrom: T,
+    vararg madeFrom: T,
     block: T.() -> String? = { null }
-  ): Graph = relationship(madeFrom, alias, block)
+  ): Graph = this.copy(
+    query = query + "-[$alias:${madeFrom.map(Relationship::type).joinToString("|")}${madeFrom.first().block().body()}]->",
+    aliases = aliases + (alias to madeFrom.toList())
+  )
 
   fun <T : Relationship> relationship(
-    relationship: T,
-    alias: String = "",
+    vararg relationship: T,
     block: T.() -> String? = { null }
   ): Graph {
     return this.copy(
-      query = query + "-[$alias:${relationship.name}${relationship.block().body()}]->",
-      aliases = aliases + (alias to relationship)
+      query = query + "-[:${relationship.map(Relationship::type).joinToString("|")}${relationship.first().block().body()}]->"
```

Comment:
This line is basically identical as the one above. Can we make a function out of the duplicated code here?